File: Python/Algorithms/58.py
class Solution:
    def lengthOfLastWord(self, s: str) -> int:
        end = len(s) - 1
        while s[end] == ' ':
            end -= 1
            if end == -1:
                return 0
        # reached end of last word.
        count = 0
        while s[end] != ' ':
            end -= 1
            count += 1
            if end == -1:
                break
        return count

File: Javascript/Algorithms/58.js
/**
 * @param {string} s
 * @return {number}
 */
var lengthOfLastWord = function(s) {
  let end = s.length - 1;
  while (s[end] == ' ') {
    if (--end == -1) return 0;
  }

  // We've reached the end of the last word (or returned).
  let count = 0;
  while (s[end] != ' ') {
    count++;
    if (--end == -1) break;
  }
  return count;
};